### PR TITLE
Integrate quality gates into WinCC OA TestFramework

### PR DESCRIPTION
**Pull request by mPokornyETM**
The goal is to integrate quality checks into WinCC OA TestFramework.
It will be possible to use this tool in our intern development process as well on customer sides.

But till integrating I found some litle issues. Therefore I refactored little the ctrl code here.

TODO:
+ More documentation with examples, how to integrate it into TestFramework (but currently not sure if shall be here or in TestFramework self)
+ remove some wasted code. Currently commented out, because I am not sure if it has sideeffects or not. But I it can be safully removed.

This changes does not inculede any tests, because after merge it will be used in our process for 3.19 and 3.20 version, therefor it will be tested indirect.

### CODE REVIEW

**Comment by Daradiridatumtarides** [comment]
Endlich tut sich wieder was in dem Repo!
Greetings from the safe side of life:
https://www.keba.com/de/industrial-automation/products/safety-technology/kesafe-sicherheitstechnik

**Comment by mPokornyETM** [comment]
> Endlich tut sich wieder was in dem Repo! Greetings from the safe side of life: https://www.keba.com/de/industrial-automation/products/safety-technology/kesafe-sicherheitstechnik

After this change will be integrated in our dev-process, and I think the findings leads to new changes also here ;-)

File: WinCCOA_QualityChecks/msg/de_AT.utf8/QgStaticCheck_OverloadedFiles.cat [hunk 1]
```diff
@@ -1,7 +1,7 @@
-assert.isCheater,Is file overloaded
+assert.isCheater,Check if is the file $location overloaded.
```
Comment (dhoegerlETM) [inline WinCCOA_QualityChecks/msg/de_AT.utf8/QgStaticCheck_OverloadedFiles.cat:1]: Check if file $location is overloaded.

File: WinCCOA_QualityChecks/msg/de_AT.utf8/QgStaticCheck_OverloadedFiles.cat [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 reason.isCheater,It is not allowed to overwrite WinCCOA-license check files: $file.name
 filesList,Files
-assert.isOverloaded,Is file overloaded
+assert.isOverloaded,Check if is the file $location overloaded.
```
Comment (dhoegerlETM) [inline WinCCOA_QualityChecks/msg/de_AT.utf8/QgStaticCheck_OverloadedFiles.cat:4]: Check if file $location is overloaded.

File: WinCCOA_QualityChecks/msg/de_AT.utf8/QgStaticCheck_OverloadedFiles.cat [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
 reason.isOverloaded,The file '$file.name' is overloaded from '$file.isOverloadedFrom'.
-assert.isOverloadedAllowed,Is file overloaded
+assert.isOverloadedAllowed,Check if is the file $location overloaded (allowed overloads).
```
Comment (dhoegerlETM) [inline WinCCOA_QualityChecks/msg/de_AT.utf8/QgStaticCheck_OverloadedFiles.cat:6]: Check if file $location is overloaded (allowed overloads).

File: WinCCOA_QualityChecks/msg/en_US.utf8/QgStaticCheck_OverloadedFiles.cat [hunk 4]
```diff
@@ -1,7 +1,7 @@
-assert.isCheater,Is file overloaded
+assert.isCheater,Check if is the file $location overloaded.
 reason.isCheater,It is not allowed to overwrite WinCCOA-license check files: $file.name
 filesList,Files
-assert.isOverloaded,Is file overloaded
+assert.isOverloaded,Check if is the file $location overloaded.
```
Comment (dhoegerlETM) [inline WinCCOA_QualityChecks/msg/en_US.utf8/QgStaticCheck_OverloadedFiles.cat:4]: Check if file $location is overloaded.

File: WinCCOA_QualityChecks/msg/en_US.utf8/QgStaticCheck_OverloadedFiles.cat [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
 reason.isOverloaded,The file '$file.name' is overloaded from '$file.isOverloadedFrom'.
-assert.isOverloadedAllowed,Is file overloaded
+assert.isOverloadedAllowed,Check if is the file $location overloaded (allowed overloads).
```
Comment (dhoegerlETM) [inline WinCCOA_QualityChecks/msg/en_US.utf8/QgStaticCheck_OverloadedFiles.cat:6]: Check if file $location is overloaded (allowed overloads).

**Review by dhoegerlETM — CHANGES_REQUESTED** [review]

**Comment by mPokornyETM** [comment]
@dhoegerlETM may you pls check my comments too? Thx

**Review by dhoegerlETM — APPROVED** [review]

**Pull request merged**